autoload/LanguageClient.vim

```diff
@@ -1634,4 +1663,81 @@ function! LanguageClient#debugInfo(...) abort
     return LanguageClient#Call('languageClient/debugInfo', l:params, l:Callback)
 endfunction
 
+function! s:ClosePopups(...) abort
+  if s:ShouldUseFloatWindow()
+    call s:CloseFloatingHover()
+  elseif exists('*popup_clear') && s:GetVar('LanguageClient_usePopupHover', v:true)
+    call popup_clear()
+  else
+    :pclose
+  endif
+endfunction
+
+" receives the v:event from the CompleteChanged autocmd
+function! LanguageClient#handleCompleteChanged(event) abort
+  " the timers on zero are just to stop textlock from locking our changes
+  if !s:Debounce(100, 'LanguageClient#handleCompleteChanged')
+    call timer_start(0, funcref('s:ClosePopups'))
+    return
+  endif
```

Comment:
I used `s:Debounce` because it was already being used but I think we should switch to something like `timer_start(100, { -> functionToCall() })` and store the returned timer id in a dictionary keyed by function name and cancel it before calling the same function again. That allows for more precise control, I've been getting very dubious results with this and I suppose it must be either the extra cycles it has to compute to evaluate the conditions for Debounce and the call itself, or maybe calculating the diff with `reltime` just isn't a reliable enough method.